sdk/tracing/tracing-opentelemetry/src/main/java/com/azure/tracing/opentelemetry/OpenTelemetryTracer.java

```diff
@@ -73,4 +104,71 @@ public void setAttribute(String key, String value, Context context) {
     public Context setSpanName(String spanName, Context context) {
         return context.addData(OPENTELEMETRY_SPAN_NAME_KEY, spanName);
     }
+
+    @Override
+    public void end(String errorCondition, Throwable throwable, Context context) {
+        Optional<Object> spanOptional = context.getData(OPENTELEMETRY_SPAN_KEY);
+        if (!spanOptional.isPresent()) {
+            logger.warning("Failed to find span to end it.");
+            return;
+        }
+
+        Span span = (Span) spanOptional.get();
+
+        if (span.getOptions().contains(Options.RECORD_EVENTS)) {
+            span.setStatus(AmqpTraceUtil.parseErrorCondition(errorCondition, throwable));
+        }
+
+        span.end();
+    }
+
+    @Override
+    public void addLink(Context eventContext) {
+        Optional<Object> spanContextOptional = eventContext.getData(SPAN_CONTEXT);
+        Optional<Object> spanOptional = eventContext.getData(OPENTELEMETRY_SPAN_KEY); // TODO: we need this to be the parent span key
+
+        if (!spanOptional.isPresent()) {
+            logger.warning("Failed to find span to link it.");
+            return;
+        }
+        SpanContext spanContext = (SpanContext) spanContextOptional.get();
+        Span span = (Span) spanOptional.get();
+
+        span.addLink(Link.fromSpanContext(spanContext, PARENT_LINKED_SPAN));
```

Comment:
can you please leave a TODO here (or where we call this method) that once we switch to OpenTelemetry, we need to add links before span is starter (and no link type - parent - is needed)